File: crates/varde-code/tests/fixtures/python/control_error.py
# Control-flow / error fixtures: ControlFlow, Catch, Throw, plus literals,
# calls, and member accesses to keep each fixture self-describing.


def process(items):
    try:
        if len(items) == 0:
            raise ValueError("empty")
        for item in items:
            if item == 0:
                continue
            while item > 10:
                item = item - 1
                break
    except ValueError as err:
        return str(err)
    finally:
        cleanup()

    with open("data.txt") as fh:
        data = fh.read()

    if data:
        return "ok"
    return None

File: crates/varde-code/tests/fixtures/ruby/control_error.rb
# Control-flow / error fixtures: ControlFlow (if/while/case/begin/return/break/
# next), Catch (rescue), and Throw (raise), plus calls and member accesses to
# keep the fixture self-describing.

def process(items)
  begin
    raise ArgumentError, "empty" if items.empty?
    items.each do |item|
      next if item.zero?
      while item > 10
        item -= 1
        break
      end
    end
  rescue ArgumentError => err
    return err.message
  ensure
    cleanup
  end

  case items.size
  when 0
    "none"
  else
    "some"
  end
end
